### PR TITLE
Remove LINQ and reflection from EasingTypeConverter

### PR DESCRIPTION
**Pull request by imememani**
### Description of Change

Clean up (using switch in place of if chain) for better readability and used more up to date .NET syntax as well as removing the usage of LINQ.

### Issues Fixed

N/A

Fixes #

N/A

### CODE REVIEW

**Review by hartez — CHANGES_REQUESTED** [review]

**Comment by affableashish** [comment]
[Switch expressions](https://learn.microsoft.com/en-us/dotnet/csharp/language-reference/operators/switch-expression) are much cleaner IMHO.

I recommend changing your switch statements to:

```c#
return strValue switch
{
    _ when Compare(nameof(Linear)) => Linear,
    _ when Compare(nameof(SinIn)) => SinIn,
    //  and so on...
    _ => "put default value here"
};
```

And add this local method for case insensitive comparison:
```c#
bool Compare(string right) => string.Equals(strValue, right, StringComparison.InvariantCultureIgnoreCase);
```

File: src/Core/src/Converters/EasingTypeConverter.cs [hunk 1]
```diff
@@ -25,40 +24,50 @@ public override object ConvertFrom(ITypeDescriptorContext context, CultureInfo c
...
-
-			var fallbackValue = typeof(Easing)
-				.GetTypeInfo()
-				.DeclaredFields
-				.FirstOrDefault(f => f.Name.Equals(strValue, StringComparison.OrdinalIgnoreCase))
-				?.GetValue(null);
+			switch (easing)
+			{
+				case nameof(Linear):
+					return nameof(Linear);
```
[the remark this answers is not in the document]
Comment (imememani) [reply src/Core/src/Converters/EasingTypeConverter.cs]: You are 100% correct, I've fixed the return value.

File: src/Core/src/Converters/EasingTypeConverter.cs [hunk 2]
```diff
@@ -25,40 +24,50 @@ public override object ConvertFrom(ITypeDescriptorContext context, CultureInfo c
...
-				return SpringIn;
-			if (strValue.Equals(nameof(SpringOut), StringComparison.OrdinalIgnoreCase))
-				return SpringOut;
-
-			var fallbackValue = typeof(Easing)
-				.GetTypeInfo()
-				.DeclaredFields
-				.FirstOrDefault(f => f.Name.Equals(strValue, StringComparison.OrdinalIgnoreCase))
-				?.GetValue(null);
+			switch (easing)
```
[the remark this answers is not in the document]
Comment (imememani) [reply src/Core/src/Converters/EasingTypeConverter.cs:33]: Fixed, it should now take this into account, thanks!

**Comment by imememani** [comment]
@affableashish Great suggestion, it does look much more readable and allows case to be ignored, I've pushed some changes to factor the suggestions in.

**Comment by albyrock87** [comment]
Is there a reason to not use `Enum.TryParse` (with case-insensitive bool parameter) and `ToString` instead of the two switch statements?

**Review by mattleibow — APPROVED** [review]

**Review by PureWeen — APPROVED** [review]

**Pull request merged**